src/e3response/data/qm9_nmr.py

```diff
@@ -0,0 +1,455 @@
+                isotropic,
+                anisotropy,
+                XX,
+                YX,
+                ZX,
+                XY,
+                YY,
+                ZY,
+                XZ,
+                YZ,
+                ZZ,
+                eigenvalue1,
+                eigenvalue2,
+                eigenvalue3,
+            ) = match
+
+            tensor_matrix = np.fromstring(
+                " ".join([XX, XY, XZ, YX, YY, YZ, ZX, ZY, ZZ]), dtype=float, sep=" "
+            ).reshape(3, 3)
+
+            molecule_data.append(
+                {
+                    "index": int(atom_number),
+                    "specie": atom_type,
+                    "tensor": tensor_matrix,
+                    "isotropic": float(isotropic),
+                    "anisotropy": float(anisotropy),
+                    "eigenvalues": [float(eigenvalue1), float(eigenvalue2), float(eigenvalue3)],
+                }
+            )
+
+        # final dictionary
+        molecule_data = {
+            "structure": structure,
+            "tensor": [atom["tensor"] for atom in molecule_data],
+            "isotropic": [atom["isotropic"] for atom in molecule_data],
+            "anisotropy": [atom["anisotropy"] for atom in molecule_data],
+            "eigenvalues": [atom["eigenvalues"] for atom in molecule_data],
+            "species": [atom["specie"] for atom in molecule_data],
+            "ind": list(range(len(structure))),
+            "N": len(structure),
+        }
+
+        return molecule_data
+
+    except ValueError as e:
+        _LOGGER.error(f"Error in file {log_file}: {e}")
+    except Exception as e:
+        _LOGGER.error(f"Error while elaborating file {log_file}: {e}")
+
+
+def get_structure_and_data_from_log(log_path: pathlib.Path) -> Optional[ase.Atoms]:
+    _LOGGER.info("Parsing Gaussian .log file: %s", log_path)
+
+    try:
+        molecule_data = create_molecule_data(log_path)
+        if molecule_data is None:
+            _LOGGER.warning("No valid structure in %s", log_path.name)
+            return None
+
+        atoms = pymatgen.io.ase.AseAtomsAdaptor.get_atoms(molecule_data["structure"])
+
+        ind = molecule_data["ind"]
+        n_atoms = molecule_data["N"]
+
+        tensors = np.zeros((n_atoms, 3, 3))
+        tensors[ind] = molecule_data["tensor"]
+
+        atoms.arrays["NMR_tensors"] = tensors
+        atoms.arrays["ind"] = np.array(ind)
+        atoms.arrays["N"] = np.array(n_atoms)
+        atoms.arrays["species"] = np.array(molecule_data["species"])
+        atoms.arrays["isotropic"] = np.array(molecule_data["isotropic"])
+        atoms.arrays["anisotropy"] = np.array(molecule_data["anisotropy"])
+        atoms.arrays["eigenvalues"] = np.array(molecule_data["eigenvalues"])
+
+        # print(atoms.arrays["anisotropy"])
+
+        return atoms
+
+    except Exception as e:
+        _LOGGER.error("Parsing error for %s: %s", log_path, e)
+        return None
+
+
+class QM9NmrDataModule(reax.DataModule):
```

Comment:
This should be `Qm9NmrDataModule` to stick to the 'camel case' convention where acronyms only have their first letter capitalized.